kuma/core/middleware.py

```diff
@@ -44,7 +54,9 @@ def process_request(self, request):
             if query_string:
                 full_path = '%s?%s' % (full_path, query_string)
 
-            response = HttpResponsePermanentRedirect(full_path)
+            # Never use HttpResponsePermanentRedirect here.
+            # Its a temporary redirect and should return with http 302, not 301
+            response = HttpResponseRedirect(full_path)
```

Comment:
This change is for a separate issue, and I'm not sure I agree with your assertion that these should be temporary redirects.  Please move this to a new PR.